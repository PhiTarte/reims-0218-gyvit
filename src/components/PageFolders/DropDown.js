import React from "react"
import {
  ButtonDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Button
} from "reactstrap"

export default class DropDown extends React.Component {
  constructor(props) {
    super(props)

    this.toggle = this.toggle.bind(this)
    this.state = {
      dropdownOpen: false
    }
  }

  toggle() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    })
  }

  render() {
    return (
      <ButtonDropdown
        style={{ marginTop: "30px" }}
        isOpen={this.state.dropdownOpen}
        toggle={this.toggle}
      >
        <DropdownToggle color="white">
          <img
            src="Assets/icon_dots_more.png"
            alt="Button Dropdown"
            aria-hidden="true"
          />
        </DropdownToggle>
        <DropdownMenu>
          <div>
            <a
              style={{
                color: "black"
              }}
              className="dropdown-item"
              href=""
            >
              <img
                style={{
                  marginRight: "6%"
                }}
                src="Assets/icon_share_dropdown.png"
                alt=""
                aria-hidden="true"
              />
              <span>Share</span>
            </a>
            <a
              style={{
                color: "black"
              }}
              className="dropdown-item"
              href=""
            >
              <img
                style={{
                  marginRight: "8%"
                }}
                src="Assets/icon_create_public_access.png"
                alt=""
                aria-hidden="true"
              />
              <span>Create public access</span>
            </a>
            <a
              style={{
                color: "black"
              }}
              className="dropdown-item"
              href=""
            >
              <img
                style={{
                  marginRight: "8%"
                }}
                src="Assets/icon_create_private_access.png"
                alt=""
                aria-hidden="true"
              />
              <span>Create private access</span>
            </a>
            <a
              style={{
                color: "black"
              }}
              className="dropdown-item"
              href=""
            >
              <img
                style={{
                  marginRight: "8%"
                }}
                src="Assets/icon_informations.png"
                alt=""
                aria-hidden="true"
              />
              <span>Informations</span>
            </a>
          </div>
          <DropdownItem divider />
          <div>
            <a
              style={{
                color: "black"
              }}
              className="dropdown-item"
              href=""
            >
              <span>Rename</span>
            </a>
            <Button
              style={{
                color: "black"
              }}
              className="dropdown-item"
              onClick={() => {
                console.log(this.props.dirId)
              }}
            >
              <span>Delete</span>
            </Button>
          </div>
        </DropdownMenu>
      </ButtonDropdown>
    )
  }
}
