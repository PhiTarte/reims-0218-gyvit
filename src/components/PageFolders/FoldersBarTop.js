import React from "react"
import {
  Col,
  Navbar,
  Nav,
  NavbarBrand,
  NavbarToggler,
  Collapse,
  NavItem,
  Button,
  Row,
  Container
} from "reactstrap"
import ModalCreateDirContainer from "../../containers/ModalCreateDirContainer"
import "./styles-page-folders/folderBarTop.css"

const FoldersBarTop = () => {
  return (
    <Container>
      <ModalCreateDirContainer />
      <Row>
        <Col xs="12">
          <Navbar
            expand="xs"
            style={{
              marginTop: "54px"
            }}
          >
            <NavbarBrand>Folders</NavbarBrand>
            <NavbarToggler />
            <Collapse navbar>
              <Nav className="ml-auto" navbar>
                <NavItem>
                  <Button className="ButtonFoldersBarTop">+</Button>
                </NavItem>
              </Nav>
            </Collapse>
          </Navbar>
        </Col>
      </Row>
    </Container>
  )
}
export default FoldersBarTop
