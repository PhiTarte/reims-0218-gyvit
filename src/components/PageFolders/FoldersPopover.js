/* eslint react/no-multi-comp: 0, react/prop-types: 0 */
import React from "react"
import { Button, Popover, PopoverHeader, PopoverBody } from "reactstrap"

export default class FoldersPopover extends React.Component {
  constructor(props) {
    super(props)

    this.togglePop = this.togglePop.bind(this)
    this.state = {
      popoverOpen: false
    }
  }

  togglePop() {
    setInterval(() => {
      this.setState({ popoverOpen: !this.state.popoverOpen })
    }, 2000)
  }

  render() {
    return (
      <div>
        <Button id="Popover1" onClick={this.togglePop}>
          Launch Popover
        </Button>
        <Popover
          placement="bottom"
          isOpen={this.state.popoverOpen}
          target="Popover1"
          toggle={this.toggle}
        >
          <PopoverHeader>Popover Title</PopoverHeader>
          <PopoverBody>
            Sed posuere consectetur est at lobortis. Aenean eu leo quam.
            Pellentesque ornare sem lacinia quam venenatis vestibulum.
          </PopoverBody>
        </Popover>
      </div>
    )
  }
}
