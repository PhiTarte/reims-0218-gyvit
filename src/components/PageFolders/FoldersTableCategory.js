import React from "react"
import PropTypes from "prop-types"

const FoldersTableCategory = ({ name = "", direction = null }) => {
  let source, alter
  switch (direction) {
    case "asc":
      source = process.env.PUBLIC_URL.concat("Assets/icon_arrow_down.png")
      alter = "down"
      break
    case "desc":
      source = process.env.PUBLIC_URL.concat("Assets/icon_arrow_up.png")
      alter = "up"
      break
    default:
      source = process.env.PUBLIC_URL.concat("Assets/icon_double_arrow.png")
      alter = "upDown"
      break
  }
  return (
    <div>
      <span>{name + "    "}</span>
      <img src={source} alt={alter} />
    </div>
  )
}

FoldersTableCategory.propTypes = {
  name: PropTypes.string.isRequired,
  direction: PropTypes.string
}

export default FoldersTableCategory
