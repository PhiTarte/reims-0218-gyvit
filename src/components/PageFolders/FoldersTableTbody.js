import React from "react"
import PropTypes from "prop-types"
import RowDir from "./RowDir"
import RowFile from "./RowFile"

const FoldersTableTbody = ({
  files = [],
  dirs = [],
  onDirclick,
  onBackclick,
  parent = {}
}) => {
  return (
    <tbody>
      {parent._id && (
        <button type="button" onClick={() => onBackclick(parent._id)}>
          {"..."}
        </button>
      )}
      {dirs.map(dir => {
        return <RowDir {...dir} onDirclick={onDirclick} />
      })}
      {files.map(file => {
        return <RowFile {...file} />
      })}
    </tbody>
  )
}

FoldersTableTbody.propTypes = {
  files: PropTypes.array,
  dir: PropTypes.array
}

export default FoldersTableTbody
