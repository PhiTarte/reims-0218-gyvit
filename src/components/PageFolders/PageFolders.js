import React from "react"
import { Container, Row, Col } from "reactstrap"
import NavbarTop from "../NavbarTop/NavbarTop"
import NavbarLeft from "../NavbarLeft/NavbarLeft"
import FoldersBarTopContainer from "../../containers/FoldersBarTopContainer"
import ModalErrorContainer from "../../containers/ModalErrorContainer"

const PageFolders = () => (
  <Container fluid>
    <Row>
      <Col
        xs="2"
        style={{
          paddingLeft: "0%",
          paddingRight: "0%",
          marginLeft: 0,
          marginRight: 0
        }}
      >
        <NavbarLeft />
      </Col>
      <Col
        xs="10"
        style={{
          paddingLeft: "0%",
          paddingRight: "0%",
          marginLeft: 0,
          marginRight: 0
        }}
      >
        <NavbarTop />
        <FoldersBarTopContainer />
        <ModalErrorContainer />
      </Col>
    </Row>
  </Container>
)

export default PageFolders
