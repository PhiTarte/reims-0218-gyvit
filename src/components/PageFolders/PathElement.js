import React from "react"

const PathElement = ({ name }) => {
  return (
    <span>
      <img
        style={{ width: "0.2em", margin: "0.4em" }}
        src={process.env.PUBLIC_URL.concat("Icons/icon_arrow_right.png")}
        alt="icon right arrow"
      />
      <span>{name}</span>
    </span>
  )
}

export default PathElement
