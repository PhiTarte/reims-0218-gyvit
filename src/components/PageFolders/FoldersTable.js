import React from "react"
import { Table, Container } from "reactstrap"
import PropTypes from "prop-types"
import FoldersTableTbody from "./FoldersTableTbody"
import FoldersTableCategory from "./FoldersTableCategory"

const FoldersTable = ({
  files = [],
  dirs = [],
  parent = {},
  onDirclick,
  onBackclick
}) => (
  <div>
    <Container
      style={{
        padding: "15px",
        width: "100%",
        height: "650px",
        overflow: " auto"
      }}
    >
      <Table borderless striped>
        <thead>
          <tr>
            <th>
              <FoldersTableCategory name="Name  " direction={null} />
            </th>
            <th>
              <FoldersTableCategory name="Last update  " direction={null} />
            </th>
            <th>
              <FoldersTableCategory name="Users  " direction={"asc"} />
            </th>
            <th />
          </tr>
        </thead>
        <FoldersTableTbody
          onDirclick={onDirclick}
          onBackclick={onBackclick}
          files={files}
          parent={parent}
          dirs={dirs}
        />
      </Table>
    </Container>
  </div>
)

FoldersTable.propTypes = {
  files: PropTypes.array,
  dir: PropTypes.array
}

export default FoldersTable
