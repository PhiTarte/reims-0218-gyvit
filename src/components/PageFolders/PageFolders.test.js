import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import PageFolders from "./PageFolders"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<PageFolders />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders link html", () => {
  const tree = renderer.create(<PageFolders />).toJSON()
  expect(tree).toMatchSnapshot()
})
