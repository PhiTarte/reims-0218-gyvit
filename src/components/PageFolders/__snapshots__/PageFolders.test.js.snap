// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <ul
            className="nav flex-column"
            expand="md"
            style={
              Object {
                "backgroundColor": "#231b56",
                "color": "white",
                "height": "1024px",
                "width": "220px",
              }
            }
          >
            <a
              className="navbar-brand"
              href="/"
              style={
                Object {
                  "backgroundColor": "#181048",
                  "color": "white",
                  "fontWeight": "bold",
                  "height": "90px",
                  "textAlign": "center",
                  "width": "220px",
                }
              }
            >
              <p>
                GYVIT
              </p>
            </a>
            <li
              className="text-center nav-item"
            />
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-md-10"
    >
      <div
        className="container"
      >
        <div
          className="row"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "border": "solid 1px #E5E8F4",
              "height": "90px",
            }
          }
        >
          <div
            className="col-7"
          >
            <div>
              <div
                className="form-group"
              >
                <span
                  className="fa fa-search"
                />
                <input
                  className="form-control"
                  id="exampleSearch"
                  name="search"
                  placeholder="Search..."
                  style={
                    Object {
                      "backgroundColor": "#fbfcfd",
                      "border": "none",
                      "borderRadius": "2px",
                      "height": "46px",
                      "marginLeft": "6px",
                      "marginTop": "22px",
                    }
                  }
                  type="search"
                />
              </div>
            </div>
          </div>
          <div>
            <img
              alt="iconNotif"
              src="img/icon_notification.png"
              style={
                Object {
                  "marginTop": "37px",
                }
              }
            />
            <span
              className="badge badge-secondary badge-pill"
              style={
                Object {
                  "backgroundColor": "#ff007e",
                  "height": "20px",
                  "marginTop": "-50px",
                  "width": "35px",
                }
              }
            >
              0
            </span>
          </div>
          <div
            className="col-3"
          >
            <div>
              <div
                className="media"
              >
                <div
                  className="media"
                  href="#"
                >
                  <img
                    alt=""
                    className="rounded-circle media-object"
                    src="img/kevinMarlot.jpeg"
                    style={
                      Object {
                        "height": "40px",
                        "marginTop": "25px",
                        "width": "40px",
                      }
                    }
                  />
                </div>
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                    style={
                      Object {
                        "color": "#231b56",
                        "fontSize": "16px",
                        "marginLeft": "21px",
                        "marginTop": "28px",
                      }
                    }
                  >
                    <span>
                      
                    </span>
                     
                    <span>
                      
                    </span>
                  </h4>
                  <div
                    className="media"
                    style={
                      Object {
                        "color": "#372c78",
                        "fontSize": "14px",
                        "marginLeft": "21px",
                        "marginTop": "5px",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <nav
            className="navbar navbar-expand"
            style={
              Object {
                "marginTop": "54px",
              }
            }
          >
            <a
              className="navbar-brand"
            >
              Folders
            </a>
            <button
              className="navbar-toggler"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              className="collapse navbar-collapse"
              style={Object {}}
            >
              <ul
                className="ml-auto navbar-nav"
              >
                <li
                  className="nav-item"
                >
                  <button
                    className="btn btn-secondary"
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#725fe3",
                        "borderRadius": "50%",
                      }
                    }
                    type={undefined}
                  >
                    +
                  </button>
                </li>
              </ul>
            </div>
          </nav>
        </div>
      </div>
      <div>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th />
              <th>
                <div>
                  Name
                  <img
                    alt="upDown"
                    src="Assets/icon_double_arrow.png"
                  />
                </div>
              </th>
              <th>
                <div>
                  Last update
                  <img
                    alt="upDown"
                    src="Assets/icon_double_arrow.png"
                  />
                </div>
              </th>
              <th>
                <div>
                  Users
                  <img
                    alt="down"
                    src="Assets/icon_arrow_down.png"
                  />
                </div>
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
